### ISSUE TITLE
Kolmogorov-Distance as quantitative check for inference results

### ISSUE THREAD

**Issue by TheVincentWagner** [Systems-Theory-in-Systems-Biology/EPI#104]
One intuitive way to measure the discrepancy between 2 samples/distributions is the Kolmogorov distance.
It would be very nice to have this feature in eulerpi. 

One implementation is readily available from scipy:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ks_2samp.html

I propose to use it only for 1D marginals, as it becomes computationally demanding in higher dimensions. 
In principle, we could also directly employ the corresponding Kolmogorov-Smirnov test that characterizes significant differences between the two samples. 
Please keep in mind, that this test is only capable of indicating discrepancies!
Also, testing on all marginals independently introduces the multiple testing problem. 
I would therefor propose to 

1) Use very clear wording regarding the interpretation of the test's outcome.
2) Print a disclaimer whenever the test is multiply used (= whenever the dimension is >1)
3) Do not report test results beyond ~5 dimensions at all.

Do you agree, @shoepfl ?

**Comment by shoepfl** [comment]
Hi,
yes, I like the idea of the Kolmogorov distance to get an idea about how close two samples are.

I also see the point for testing, and I think it makes sense for low dimensions as it is a clear statement compared to a "kind of arbitrary" distance value.

Regarding multiple testing, we should state 
1st, it is impossible to prove that the two samples are equal, the only possible statement is that they are not significantly different. This does not allow the reverse conclusion that they are the same (as we have a frequentist setting), but at least shows that they are not so different that it would be significant.

2nd As we do not want a significant result here for any dimension, this gets harder to achieve for more dimensions. So its the other way around as it would be in traditional multiple testing. So you could also test for more than 5 dimensions, but the alpha error accumulates there.

3rd As we kind of misuse testing in a frequentist setting (we would like to accept the null hypotheses, which is impossible), I would suggest we think about a different term than testing, so rather speak of assessment (or do you have any suggestions on this @TheVincentWagner ).

**Comment by TheVincentWagner** [comment]
We test for dissimilarity, don't we? So in my eyes, we could just call it like this. Alternatively, if we would like to formulate it more positively, I quite like your proposal of a similarity assessment. I slightly prefer the first option, as it is mathematically well-defined. But I am not super strong in this opinion. How do you feel, @shoepfl ?

**Comment by shoepfl** [comment]
I agree, dissimilarity is exactly what we look for. So lets call it like that
